### PR TITLE
Feature/fast tf idf

### PR DESCRIPTION
**Pull request by aCampello**
## Description

This PR adds the possibility of using the parallel version of the tf-idf vectoriser and makes MeshXLinear sklearn compatible.

With this, I believe it will unlock us some experiment like increasing max_features and ngram #149 .

It also adds a small script to notify slack after training finished.

[Notion card](https://www.notion.so/wellcometrust/dc03328f311d4676b497ec7cd6f51042?v=e1a6758d760d4689b24356826b2d3541&p=70909bf9f59749d3a68cae865dc120b2)

## Checklist

- [x] Linked to Notion or GitHub issue
- [x] Added tests (Modified tests so it runs)
- [x] Updated README
- [ ] DVC up to date: 🚫 

## Release checklist

- [ ] DVC repro up to date
- [ ] Models synced in S3
- [ ] Version updated
- [ ] CHANGELOG.md updated
- [ ] Model and package version aligned

To release:

* `make build`
* `make deploy`

### CODE REVIEW

File: .envrc
Comment (nsorros) [inline .envrc:9]: did you mean to share this? better to populate the `.envrc.template` and let the user have their own `.envrc`. also note that if you populate the values git will detect a change if the file is git tracked.

File: grants_tagger/models/mesh_xlinear.py
Comment (nsorros) [inline grants_tagger/models/mesh_xlinear.py:138]: not that this is a bad strategy but what would be even better is to save the params in a `meta.json` and initialize from there so that you don't need to pass params when loading the model. otherwise a user might save a vectorizer and then try to load without knowing which vectorizer was used. the information should ideally be stored somewhere. `spacy` uses a `meta.json` as well. I have short of tried following that route but it is not complete yet in other models so its optional

**Review by nsorros — APPROVED** [review]
LGTM 🚀 I can't believe that a parallel tfidf was hiding under our nose. Have a think about whether params should be saved in the model_path somehow.

**Comment by nsorros** [comment]
⚠️ Wait till the release goes out first #143 @aCampello

File: .envrc
Comment (aCampello) [reply .envrc:9]: Oopsy. Fixed https://github.com/wellcometrust/grants_tagger/pull/147/commits/a435335ec02be91ca14630551f5e84063fc19325 and added a gitignore too

File: grants_tagger/models/mesh_xlinear.py
Comment (aCampello) [reply grants_tagger/models/mesh_xlinear.py:138]: Yep, meta route is definitely better. Let's add this to improvements!

**Comment by nsorros** [comment]
@aCampello do you want me to take over and run `dvc repro`? You can also do btw before you merge. Or we can merge your other PR as well before running.

**Comment by aCampello** [comment]
Maybe we can merge this, and then when I submit the optimisation one (which will be today), we can run dvc repro?

**Comment by nsorros** [comment]
> Maybe we can merge this, and then when I submit the optimisation one (which will be today), we can run dvc repro?

Sounds good

**Comment by nsorros** [comment]
🎉

**Pull request merged**
